### PR TITLE
BaSP-TG-28: Admins. 

### PR DESCRIPTION
**Pull request by lucianofrailelopez**
1. Show admin list.
2. Delete admin.
3. Add admin.
4. Edit admin.

### CODE REVIEW

**Review by iaracriscenti — COMMENTED** [review]
Some recommended changes to make.

File: src/Components/Admins/App.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React, { useState, useEffect } from 'react';
+import List from './List/List';
+
+const App = () => {
+  const [admins, setAdmins] = useState([]);
+
+  useEffect(async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}api/admins`);
```
Comment (NicolasFontana) [inline src/Components/Admins/App.jsx:9]: The right resource is https://alexis-trackgenix-server.vercel.app. Should put {process.env.REACT_APP_API_URL}/api/employees/${_id}

File: src/Components/Admins/Edit/index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    email: '',
+    password: '',
+    active: ''
+  });
+
+  useEffect(async () => {
+    try {
+      const params = new URLSearchParams(window.location.search);
+      const adminID = params.get('id');
+      const getAdmin = await fetch(`${process.env.REACT_APP_API_URL}api/admins/${adminID}`);
```
Comment (NicolasFontana) [inline src/Components/Admins/Edit/index.js:18]: The right resource is https://alexis-trackgenix-server.vercel.app/. Should put {process.env.REACT_APP_API_URL}/api/employees/${_id}

File: src/Components/Admins/Edit/index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      console.error(error);
+      alert(error);
+    }
+  }, []);
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    const params = new URLSearchParams(window.location.search);
+    const AdminID = params.get('id');
+    fetch(`${process.env.REACT_APP_API_URL}api/admins/${AdminID}`, {
```
Comment (NicolasFontana) [inline src/Components/Admins/Edit/index.js:37]: The right resource is https://alexis-trackgenix-server.vercel.app. Should put {process.env.REACT_APP_API_URL}/api/employees/${_id}

File: src/Components/Admins/App.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      console.error(error);
+    }
+  }, []);
+
+  const deleteAdmins = async (_id) => {
+    try {
+      const response = await fetch(`http://localhost:8000/api/admins/${_id}`, {
+        method: 'DELETE'
+      });
+      console.log('response', response);
```
Comment (NicolasFontana) [inline src/Components/Admins/App.jsx:22]: console.log

**Review by NicolasFontana — CHANGES_REQUESTED** [review]

**Review by iaracriscenti — APPROVED** [review]

**Review by NicolasFontana — APPROVED** [review]
🚀

**Comment by LeonelListro** [comment]
## Testing Results
Tested locally on https://alexis-trackgenix-app-git-feature-tg-28admin-basp-m2022.vercel.app/

Test Results: Failed ⚠️

Bugs found:

1. The alert message depoyed when click delete icon is wrong, should be "Are you sure you want to delete this admin" and it says "Are you sure you want to delete this super admin".

image

2. This button should not exist, it has the same function of the cancel button.

image

3. According to the convencions of "Material Design (https://material.io/)", the cancel button should be on left side, and the save button should be on the right side with a different text (could be "create").

dsBuffer bmp

4. The active field is not required to create a new admin and it should be required.

image

5. The delete text message should be "Admin deleted" and below "ID: number of id", no need for the ID to appear twice

image

6. The error message when you try to edit the name or/and the last name with numbers is difficult to understand because it shows a regex validation.

image

**Review by LeonelListro — CHANGES_REQUESTED** [review]
Please refer to: https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/9#issuecomment-1145531563

**Comment by AlexisPuchetaRR** [comment]
Leeeendo testeo se mando Leo 👏

**Review by LeonelListro — APPROVED** [review]
Test Result: Approve 👍

**Comment by ArianaMaldonado** [comment]
Tested locally on https://alexis-trackgenix-app-git-feature-tg-28admin-basp-m2022.vercel.app/

**Test Results: Failed :warning:**

Bugs found:
1. When creating/editing an Admin, we should be redirected to the Admin list again.
Screenshot from 2022-06-03 19-05-57

2. When we edit an Admin which 'Active' property is _true_, if we click on 'edit' without changing the _'Active'_ property, it changes to _false_ by default.
Screenshot from 2022-06-03 19-14-41
Screenshot from 2022-06-03 19-17-59

3. Wording bug. The message should be: 'Admin **deleted** ....'
Screenshot from 2022-06-03 19-23-23

4. Should password be visible?
Screenshot from 2022-06-03 19-28-38

**Review by ArianaMaldonado — CHANGES_REQUESTED** [review]
Changes requested. Please, refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/9#issuecomment-1146414215

**Review by ArianaMaldonado — APPROVED** [review]

**Pull request merged**